storage/cloud-client/encryption_test.py

```diff
@@ -54,18 +56,32 @@ def test_upload_encrypted_blob():
         )
 
 
-@pytest.fixture
+@pytest.fixture(scope="module")
 def test_blob():
     """Provides a pre-existing blob in the test bucket."""
     bucket = storage.Client().bucket(BUCKET)
+    blob_name = "test_blob_{}".format(uuid.uuid4().hex)
     blob = Blob(
-        "encryption_test_sigil",
+        blob_name,
         bucket,
         encryption_key=TEST_ENCRYPTION_KEY_DECODED,
     )
     content = "Hello, is it me you're looking for?"
     blob.upload_from_string(content)
-    return blob.name, content
+
+    yield blob.name, content
+
+    try:
+        blob.delete()
+    except NotFound as e:
+        # For the case that the rotation succeeded.
+        print("Ignoring 404, detail: {}".format(e))
+        blob = Blob(
+            blob_name,
+            bucket,
+            encryption_key=TEST_ENCRYPTION_KEY_2_DECODED
```

Comment:
I'm surprised you need to pass in the encryption key to delete the object. Could you provide context on how you arrived to this solution?

Reply:
I believe that is something that already exists in the sample

Reply:
@frankyn 

I added some comments. I just got `NotFound` and realized you need the key to delete the blob.

Reply:
Without the try-except, it is always fail with the following error:

google.api_core.exceptions.NotFound: 404 DELETE https://storage.googleapis.com/storage/v1/b/tmatsuo-test-bucket/o/test_blob_7188d5dc44764ef7a6bc19a815385daf?generation=1588127655320921: No such object: tmatsuo-test-bucket/test_blob_7188d5dc44764ef7a6bc19a815385daf